crates/bevy_reflect/src/impls/std.rs

```diff
@@ -1039,6 +1039,131 @@ impl FromReflect for Cow<'static, str> {
     }
 }
 
+impl<T: FromReflect + Clone + Send + Sync> Array for Cow<'static, [T]> {
+    fn get(&self, index: usize) -> Option<&dyn Reflect> {
+        self.as_ref().get(index).map(|x| x as &dyn Reflect)
+    }
+
+    fn get_mut(&mut self, index: usize) -> Option<&mut dyn Reflect> {
+        self.to_mut().get_mut(index).map(|x| x as &mut dyn Reflect)
+    }
+
+    fn len(&self) -> usize {
+        self.as_ref().len()
+    }
+
+    fn iter(&self) -> ArrayIter {
+        ArrayIter::new(self)
+    }
+
+    fn drain(self: Box<Self>) -> Vec<Box<dyn Reflect>> {
+        self.into_owned()
+            .into_iter()
+            .map(|x| Box::new(x) as Box<dyn Reflect>)
+            .collect()
+    }
+}
+
+impl<T: FromReflect + Clone + Send + Sync> Reflect for Cow<'static, [T]> {
+    fn type_name(&self) -> &str {
+        std::any::type_name::<Self>()
+    }
+
+    fn get_type_info(&self) -> &'static TypeInfo {
+        <Self as Typed>::type_info()
+    }
+
+    fn into_any(self: Box<Self>) -> Box<dyn Any> {
+        self
+    }
+
+    fn as_any(&self) -> &dyn Any {
+        self
+    }
+
+    fn as_any_mut(&mut self) -> &mut dyn Any {
+        self
+    }
+
+    fn into_reflect(self: Box<Self>) -> Box<dyn Reflect> {
+        self
+    }
+
+    fn as_reflect(&self) -> &dyn Reflect {
+        self
+    }
+
+    fn as_reflect_mut(&mut self) -> &mut dyn Reflect {
+        self
+    }
+
+    fn apply(&mut self, value: &dyn Reflect) {
+        let value = value.as_any();
+        if let Some(value) = value.downcast_ref::<Self>() {
+            *self = value.clone();
+        } else {
+            panic!("Value is not a {}.", std::any::type_name::<Self>());
+        }
+    }
+
+    fn set(&mut self, value: Box<dyn Reflect>) -> Result<(), Box<dyn Reflect>> {
+        *self = value.take()?;
+        Ok(())
+    }
+
+    fn reflect_ref(&self) -> ReflectRef {
+        ReflectRef::Array(self)
+    }
+
+    fn reflect_mut(&mut self) -> ReflectMut {
+        ReflectMut::Array(self.to_mut())
+    }
+
+    fn reflect_owned(self: Box<Self>) -> ReflectOwned {
+        ReflectOwned::Array(Box::new(self.to_vec()))
+    }
+
+    fn clone_value(&self) -> Box<dyn Reflect> {
+        Box::new(self.clone())
+    }
+
+    fn reflect_hash(&self) -> Option<u64> {
+        crate::array_hash(self)
+    }
+
+    fn reflect_partial_eq(&self, value: &dyn Reflect) -> Option<bool> {
+        crate::array_partial_eq(self, value)
+    }
+}
+
+impl<T: FromReflect + Clone + Send + Sync> Typed for Cow<'static, [T]> {
+    fn type_info() -> &'static TypeInfo {
+        static CELL: NonGenericTypeInfoCell = NonGenericTypeInfoCell::new();
+        CELL.get_or_set(|| TypeInfo::Array(ArrayInfo::new::<Self, T>(0)))
```

Comment:
Hm, it's kinda weird that we'd set the capacity to `0` here. I wonder if there's a better way of doing this? 🤔

Reply:
Yeah I was going to ask about that. I thought a slice type didn't have a capacity, unless we consider whatever the hardware's [max isize](https://doc.rust-lang.org/stable/reference/types/numeric.html#machine-dependent-integer-types) as a capacity.

Trying to implement this almost made me feel like it'd be more natural to be returning a Reflect***::List instead of the Array case. I haven't tried that yet, but that could keep us from deciding on a random capacity.

Reply:
Can you insert/remove data on a slice? If so then yeah `List` would be better. Otherwise, I’m not sure of the best approach...

Reply:
You can't insert/remove data on a regular slice, but you can on a Cow slice by doing a `to_mut` and then inserting/removing.

Reply:
Basically a `Cow<'static, [T]>` is a `Vec<T>` that can optionally store ALL of its data in static memory. Therefore it's closer to being a `Vec<T>` than `[T; N]`, so it should be treated as a `List` in reflection.

Reply:
@ItsDoot I added your implementation of ```List``` for ```Cow<'static, [T]>``` from #7455 and switched the ```Array``` reflections to use the ```List``` implementation. By the way, I'm sorry if I hijacked your issue. I should have left a comment asking to claim it first.